src/transformState.js

```diff
@@ -61,8 +61,27 @@
  * @param {Object} state
  * @param {Object[]} transforms
  */
+
 function transformState(state, transforms) {
-  // write code here
+  for (const key of transforms) {
+    switch (key.operation) {
+      case 'addProperties':
+        Object.assign(state, key.properties);
+        break;
+
+      case 'removeProperties':
+        for (const i of key.properties) {
+          delete state[i];
+        }
+        break;
+
+      case 'clear':
+        for (const i in state) {
```

Comment:
```suggestion
        for (const property of key.properties) {
          delete state[property];
        }
        break;

      case 'clear':
        for (const value in state) {
```

try to use more meaningful names for variables